Assets/GameAssets/Player/PlayerScripts/PlayerInteraction.cs

```diff
@@ -0,0 +1,108 @@
+/**********************************************************************************************************************
+// File Name :         PlayerInteraction.cs
+// Author :            Alex Kalscheur
+// Creation Date :     11/10/24
+// 
+// Brief Description : Controls the interaction system on the player side
+**********************************************************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+[RequireComponent(typeof(PlayerInput))]
+
+/// <summary>
+/// Controls the interaction system on the player side
+/// </summary>
+public class PlayerInteraction : MonoBehaviour
+{
+    [Tooltip("How far can the player reach interactable objects")]
+    [SerializeField] private float _maxReach;
+    [Tooltip("Ray that is cast from camera to find interactable objects")]
+    Ray ray;
```

Comment:
I would prefer if it were specified that this is private, just for reading clarity. Also since it's private it should follow the naming scheme